From 43d5d297e84faa4897d034a7f5487a15c4637aff Mon Sep 17 00:00:00 2001
From: Michael Dege <michael.dege@renesas.com>
Date: Tue, 17 May 2022 13:12:55 +0200
Subject: [PATCH 3/3] arch: arm64: dts: renesas: r8a779f0-s4sk-prototype: Add
 rswitch node

The patch adds rswitch node channel 0/1 for S4 Starter Kit prototype.
Based on the original (and large) patch by Michael Dege.

Signed-off-by: Michael Dege <michael.dege@renesas.com>
Signed-off-by: Tam Nguyen <tam.nguyen.xa@renesas.com>
---
 .../dts/renesas/r8a779f0-s4sk-prototype.dts   | 65 +++++++++++++++++++
 1 file changed, 65 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-s4sk-prototype.dts b/arch/arm64/boot/dts/renesas/r8a779f0-s4sk-prototype.dts
index a3d6cd8c60e7..c13afece9381 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0-s4sk-prototype.dts
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-s4sk-prototype.dts
@@ -16,6 +16,7 @@ / {
 	aliases {
 		serial0 = &hscif0;
 		serial1 = &hscif1;
+		eth0	= &rswitch;
 	};
 
 	chosen {
@@ -167,6 +168,35 @@ pcie1_pins: pcie1 {
 		function = "pcie";
 	};
 
+	tsn0_pins: tsn0 {
+		mux {
+			groups = "tsn0_link", "tsn0_mdio";
+			function = "tsn0";
+			drive-strength = <18>;
+			power-source = <3300>;
+		};
+
+		pins_mdio {
+			groups = "tsn0_mdio";
+			drive-strength = <18>;
+			power-source = <3300>;
+		};
+	};
+
+	tsn1_pins: tsn1 {
+		mux {
+			groups = "tsn1_link", "tsn1_mdio";
+			function = "tsn1";
+			drive-strength = <18>;
+			power-source = <3300>;
+		};
+
+		pins_mdio {
+			groups = "tsn1_mdio";
+			drive-strength = <18>;
+			power-source = <3300>;
+		};
+	};
 };
 
 &rwdt {
@@ -231,3 +261,38 @@ &pciec1_ep {
 	pinctrl-0 = <&pcie1_pins>;
 	pinctrl-names = "default";
 };
+
+&rswitch {
+	status = "okay";
+	pinctrl-0 = <&tsn0_pins &tsn1_pins>;
+	pinctrl-names = "default";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		tsn0: port@0 {
+			reg = <0>;
+			phy-mode = "usxgmii";
+			phy-handle = <&etha0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			etha0: ethernet-phy@0 {
+				reg = <1>;
+				compatible = "ethernet-phy-ieee802.3-c45";
+			};
+		};
+		tsn1: port@1 {
+			reg = <1>;
+			phy-mode = "usxgmii";
+			phy-handle = <&etha1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			etha1: ethernet-phy@1 {
+				reg = <2>;
+				compatible = "ethernet-phy-ieee802.3-c45";
+			};
+		};
+	};
+};
-- 
2.25.1

